src/sinol_make/config/__init__.py

```diff
@@ -0,0 +1,39 @@
+import yaml, os, subprocess
+
+class Config:
+	def __init__(self):
+		config_path = os.path.join(os.path.dirname(__file__), 'config.yml')
+		if not os.path.exists(config_path):
+			open(config_path, 'w').write(yaml.dump({}))
+			self.setup()
+
+	def setup(self):
+		print("Configure your compilers:")
+		
+		self.set('c_compiler', self.__ask_value("C compiler", "gcc"))
+		self.set('cpp_compiler', self.__ask_value("C++ compiler", "g++"))
+		self.set('python_compiler', self.__ask_value("Python compiler", "python3"))
+		self.set('java_compiler', self.__ask_value("Java compiler", "javac"))
```

Comment:
Are there possible scenarios where the user would want something else?

Reply:
Yes, for example on MacOS `g++` is clang compiler. The normal GNU C++ compiler is named g++-12 for version 12 or g++-9 for version 9.

Reply:
Maybe then we can try to deduce the compiler the tool should use, instead of asking the user to provide explicitly this information? Though then it might be beneficial to overwrite something, but I would argue that it should be a rare case and an argument/flag might be better suited.